### ISSUE TITLE
CI builds

### ISSUE THREAD

**Issue by kovalikp** [nuproj/nuproj#127]
I'm trying to get CI builds running, but I'm stuck on one thing. `NuProj.ProjectSystem.14` fails to build, both with and without referencing `Microsoft.Collections.Immutable` assembly.

With `Microsoft.Collections.Immutable` reference:
https://ci.appveyor.com/project/kovalikp/nuproj/build/1.0.6#L76

Without `Microsoft.Collections.Immutable` reference:
https://ci.appveyor.com/project/kovalikp/nuproj/build/1.0.8#L73

It appears that `NuProj.ProjectSystem.14` is using `lib\Xaml2Cs.exe` instead of `lib14\Xaml2Cs.exe` for whatever reason to generate .cs files. Any idea what could be wrong? The build is working locally just fine :confused: 

The CI build is set to use Visual Studio 2015 RC environment.

**Comment by AArnott** [comment]
`NuProj.ProjectSystem.14` should be using `lib14\Xaml2cs.exe`. And it should be referencing the `System.Collections.Immutable` NuGet package *instead of* `Microsoft.Collections.Immutable`

**Comment by kovalikp** [comment]
Yes, that's what it appears to be  doing, at least according to logs.

```
[00:02:14] Target "GenerateRuleSourceFromXaml" in file "C:\projects\nuproj\src\NuProj.ProjectSystem.12\ProjectSystem.targets" from project "C:\projects\nuproj\src\NuProj.ProjectSystem.12\NuProj.ProjectSystem.12.csproj" (target "PrepareResourceNames" depends on it):
[00:02:14] Using "XamlRuleToCode" task from assembly "C:\projects\nuproj\src\NuProj.ProjectSystem.12\..\..\lib\Xaml2Cs.exe".
[00:02:14] Task "XamlRuleToCode"
[00:02:14]   Generating code for 'Rules\general.xaml'.
[00:02:14]   Generating code for 'Rules\general.browseobject.xaml'.
[00:02:14]   Generating code for 'Rules\folder.xaml'.
[00:02:14]   Generating code for 'Rules\content.xaml'.
[00:02:14]   Generating code for 'Rules\none.xaml'.
[00:02:15]   Generating code for 'Rules\scc.xaml'.
[00:02:15]   Generating code for 'Rules\ProjectReference.xaml'.
[00:02:15]   Generating code for 'Rules\ResolvedProjectReference.xaml'.
[00:02:15] Done executing task "XamlRuleToCode".
[00:02:15] Done building target "GenerateRuleSourceFromXaml" in project "NuProj.ProjectSystem.12.csproj".
```

```
[00:03:01] Target "GenerateRuleSourceFromXaml" in file "C:\projects\nuproj\src\NuProj.ProjectSystem.14\ProjectSystem.targets" from project "C:\projects\nuproj\src\NuProj.ProjectSystem.14\NuProj.ProjectSystem.14.csproj" (target "PrepareResourceNames" depends on it):
[00:03:01] Using "XamlRuleToCode" task from assembly "C:\projects\nuproj\src\NuProj.ProjectSystem.14\..\..\lib14\Xaml2Cs.exe".
[00:03:01] Task "XamlRuleToCode"
[00:03:01]   Generating code for '..\NuProj.ProjectSystem.12\Rules\content.xaml'.
[00:03:03]   Generating code for '..\NuProj.ProjectSystem.12\Rules\folder.xaml'.
[00:03:03]   Generating code for '..\NuProj.ProjectSystem.12\Rules\general.browseobject.xaml'.
[00:03:03]   Generating code for '..\NuProj.ProjectSystem.12\Rules\general.xaml'.
[00:03:03]   Generating code for '..\NuProj.ProjectSystem.12\Rules\none.xaml'.
[00:03:03]   Generating code for '..\NuProj.ProjectSystem.12\Rules\ProjectReference.xaml'.
[00:03:03]   Generating code for '..\NuProj.ProjectSystem.12\Rules\ResolvedProjectReference.xaml'.
[00:03:03]   Generating code for '..\NuProj.ProjectSystem.12\Rules\scc.xaml'.
[00:03:03] Done executing task "XamlRuleToCode".
[00:03:03] Done building target "GenerateRuleSourceFromXaml" in project "NuProj.ProjectSystem.14.csproj".
```
And yet it seems to generate `Microsoft.Collections.Immutable` for `NuProj.ProjectSystem.14`:
```
[00:03:16] obj\Release\content.cs(78,104): error CS0234: The type or namespace name 'Collections' does not exist in the namespace 'Microsoft' (are you missing an assembly reference?) [C:\projects\nuproj\src\NuProj.ProjectSystem.14\NuProj.ProjectSystem.14.csproj]
```

:unamused:

**Comment by AArnott** [comment]
Then my guess is it's a problem with the .targets files. The `GeneralRuleSourceFromXaml` task accepts parameters that specify the type names to use for several functions. Immutable collections might be one of them.

**Comment by kovalikp** [comment]
@terrajobst Can you please update appveyor settings to use VS 2015 RTM environment and use MSBuild 14.0? CI builds should be working now.

**Comment by kovalikp** [comment]
Resolved with #136
